sphinxcontrib/confluencebuilder/storage/translator.py

```diff
@@ -1145,13 +1144,9 @@ def visit_table(self, node):
         table_classes = node.get('classes', [])
         attribs = {}
 
-        # For v2 editor, if we have given explicit widths for columns in the
-        # table (e.g. CSV table), we need to apply a data table width or the
-        # editor will ignore the column-specific widths. If widths are
-        # detected, apply the default table width observed when using the v2
-        # editor.
-        if self.v2 and 'colwidths-given' in table_classes:
-            attribs['data-table-width'] = CONFLUENCE_DEFAULT_V2_TABLE_WIDTH
+        # Apply the default table width when using the v2 editor.
+        if self.v2:
+            attribs['data-table-width'] = config.confluence_default_table_width
```

Comment:
```suggestion
        # Apply the default table width when using the v2 editor.
        if self.v2 and config.confluence_default_table_width:
            attribs['data-table-width'] = config.confluence_default_table_width
        # For v2 editor, if we have given explicit widths for columns in the
        # table (e.g. CSV table), we need to apply a data table width or the
        # editor will ignore the column-specific widths. If widths are
        # detected, apply the default table width observed when using the v2
        # editor.
        elif self.v2 and 'colwidths-given' in table_classes:
            attribs['data-table-width'] = CONFLUENCE_DEFAULT_V2_TABLE_WIDTH
```

With this and the other change, you should be able to get your user-configured fixed widths from `confluence_default_table_width`. And, if `confluence_default_table_width` is not set and if `colwidths-given`, this will also avoid the need to configure a default width value for tables to allow Confluence manage this for us.